test/base/features/overview/OverviewSpec.js

```diff
@@ -0,0 +1,258 @@
+        return expectEventually(() => {
+          const newNumberOfElements = overviewContainer.get('elementRegistry').getAll().length;
+
+          return newNumberOfElements === (originalNumberOfElements + 1);
+        });
+      })
+    );
+  });
+
+
+  describe('opening element', function() {
+
+    beforeEach(bootstrapBaseModeler(diagramXML));
+
+
+    it('should open decision table when clicked', inject(async function(_parent) {
+
+      // given
+      await openView(_parent, 'decisionTable');
+      const overviewContainer = getModule(_parent, 'overviewContainer');
+      await waitForOverviewUpdate(overviewContainer);
+      const elementRegistry = overviewContainer.get('elementRegistry');
+      const decision = elementRegistry.get('season');
+      const eventBus = overviewContainer.get('eventBus');
+
+      // when
+      eventBus.fire('element.click', { element: decision });
+
+      // then
+      await expectEventually(() => {
+        const visual = elementRegistry.getGraphics(decision);
+        return visual.classList.contains('open');
+      });
+    }));
+
+
+    it('should NOT mark text annotation as can-open', inject(async function(_parent) {
+
+      // given
+      await openView(_parent, 'decisionTable');
+      const overviewContainer = getModule(_parent, 'overviewContainer');
+      await waitForOverviewUpdate(overviewContainer);
+      const elementRegistry = overviewContainer.get('elementRegistry');
+      const textAnnotation = elementRegistry.get('TextAnnotation_1');
+
+      // when
+      const visual = elementRegistry.getGraphics(textAnnotation);
+
+      // then
+      expect(visual.classList.contains('can-open')).to.be.false;
+    }));
+
+
+    it('should NOT mark decision without logic as can-open', inject(async function(_parent) {
+
+      // given
+      await openView(_parent, 'decisionTable');
+      const overviewContainer = getModule(_parent, 'overviewContainer');
+      await waitForOverviewUpdate(overviewContainer);
+      const elementRegistry = overviewContainer.get('elementRegistry');
+      const decision = elementRegistry.get('empty');
+
+      // when
+      const visual = elementRegistry.getGraphics(decision);
+
+      // then
+      expect(visual.classList.contains('can-open')).to.be.false;
+    }));
+  });
+});
+
+
+
+// helpers //////////
+function openView(dmnjs, type) {
+  const view = dmnjs.getViews().find(view => view.type === type);
+  return dmnjs.open(view);
+}
+
+function getModule(dmnjs, name) {
+  const activeViewer = dmnjs.getActiveViewer();
+
+  return activeViewer.get(name);
+}
+
+async function waitForOverviewUpdate(overviewContainer) {
+  return new Promise(resolve => {
+    overviewContainer.on('import.done', function onImport() {
+      overviewContainer.off('import.done', onImport);
+      resolve();
+    });
+  });
+}
+
+/**
+ * Wait 10 promise ticks to ensure predicate does not throw.
+ * @param {() => boolean} predicate
+ */
+async function expectEventually(predicate) {
```

Comment:
I guess there's no way to avoid this? Why not use https://testing-library.com/docs/dom-testing-library/api-async#waitfor then?

Reply:
I think so. When the view is changed, overview initiates an async import but we don't wait for the result. We could use the library you provided.

Reply:
Done (cf. https://github.com/camunda/camunda-dmn-js/pull/14/commits/41f50057eec2ca9e6e4808af8fd882aea37ea5b1).